include/mbedtls/asn1write.h

```diff
@@ -381,6 +362,31 @@ mbedtls_asn1_named_data *mbedtls_asn1_store_named_data(mbedtls_asn1_named_data *
                                                        const unsigned char *val,
                                                        size_t val_len);
 
+/**
+ * \brief                  Encode an integer into ASN1 and write it to a buffer.
+ *                         The function takes a big endian bignum buffer that
+ *                         begins at start and ends at *p and encodes it into
+ *                         a asn1 frame.
+ *                         Warning: Destructive behaviour for the buffer data!
```

Comment:
I don't understand this warning. It seems to contradict the `const` annotation on `integer`.

I would strongly expect `integer` to be const. It's an input, not a working buffer!